tests/unit/utils/metadata.spec.js

```diff
@@ -92,4 +93,11 @@ const assertMetadata = ({
 describe('Metadata', () => {
   assertMetadata(pageWithTitleAndDescription);
   assertMetadata(pageWithoutTitleOrDescription);
+
+  it('does not inherit metadata from previous pages', () => {
```

Comment:
You can also add a test case, whether it removes tags, that are not in the new page.